src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -0,0 +1,202 @@
+    return methodName(
+        Name.from(getReducedServiceName(service), collectionConfig.getEntityName(), "path"));
+  }
+
+  @Override
+  public String getStaticLangReturnTypeName(Method method, MethodConfig methodConfig) {
+    return converter.getTypeName(method.getOutputType()).getFullName();
+  }
+
+  @Override
+  public List<String> getDocLines(ProtoElement element) {
+    if (!(element instanceof Method)) {
+      return super.getDocLines(element);
+    }
+    Method method = (Method) element;
+    String text = DocumentationUtil.getDescription(method);
+    text = lowerFirstLetter(text);
+    return super.getDocLines(getApiMethodName(method) + " " + text);
+  }
+
+  @Override
+  public String getAndSavePagedResponseTypeName(
+      ModelTypeTable typeTable, TypeRef inputType, TypeRef outputType, TypeRef resourceType) {
+    String typeName = converter.getTypeNameForElementType(resourceType).getNickname();
+    int dotIndex = typeName.indexOf('.');
+    if (dotIndex >= 0) {
+      typeName = typeName.substring(dotIndex + 1);
+    }
+    return className(Name.anyCamel(typeName).join("iterator"));
+  }
+
+  private static String lowerFirstLetter(String s) {
+    if (s.length() > 0) {
+      s = Character.toLowerCase(s.charAt(0)) + s.substring(1);
+    }
+    return s;
+  }
+
+  @Override
+  public String getGrpcClientTypeName(Interface service) {
+    return converter.getTypeName(service).getNickname() + "Client";
+  }
+
+  @Override
+  public String getGrpcClientConstructorName(Interface service) {
+    return getGrpcClientTypeName(service).replace(".", ".New");
+  }
+
+  @Override
+  public String getCallSettingsTypeName(Interface service) {
+    return className(clientNamePrefix(service).join("call").join("options"));
+  }
+
+  @Override
+  public String getDefaultApiSettingsFunctionName(Interface service) {
+    return Name.from("default")
+        .join(clientNamePrefix(service))
+        .join("client")
+        .join("options")
+        .toLowerCamel();
+  }
+
+  @Override
+  public String getDefaultCallSettingsFunctionName(Interface service) {
+    return Name.from("default")
+        .join(clientNamePrefix(service))
+        .join("call")
+        .join("options")
+        .toLowerCamel();
+  }
+
+  @Override
+  public String getApiWrapperClassName(Interface service) {
+    return className(clientNamePrefix(service).join("client"));
+  }
+
+  @Override
+  public String getApiWrapperClassConstructorName(Interface service) {
+    return methodName(Name.from("new").join(clientNamePrefix(service)).join("client"));
+  }
+
+  @Override
+  public String getApiWrapperClassConstructorExampleName(Interface service) {
+    return methodName(
+        Name.from("example").join("new").join(clientNamePrefix(service)).join("client"));
+  }
+
+  @Override
+  public String getApiMethodExampleName(Interface service, Method method) {
+    return methodName(Name.from("example").join(clientNamePrefix(service)).join("client"))
+        + "_"
```

Comment:
Unsafe concatenation. The output looks like `ExampleMyType_MyMethod`, neither camel nor underscore.